pymeasure/instruments/keithley/keithley2306.py

```diff
@@ -0,0 +1,676 @@
+        ":SENS<ch>:LINT:TIME?", ":SENS<ch>:LINT:TIME %g",
+        """A floating point property that controls the long integration time
+        in seconds, which takes on values in the range of 0.850 for 60 Hz and
+         0.840 for 50 Hz up to 60. """,
+        validator=truncated_range,
+        values=[0.840, 60],
+    )
+
+    def long_integration_time_auto(self):
+        """Arranges for the instrument to control integration times. """
+        self.write(":SENS<ch>:LINT:TIME:AUTO")
+
+    long_integration_trigger_level = Instrument.control(
+        ":SENS<ch>:LINT:TLEV?", ":SENS<ch>:LINT:TLEV %g",
+        """A floating point property that controls the long integration trigger
+        level in amps, which takes values between 0 and 5. """,
+        validator=truncated_range,
+        values=[0, 5],
+    )
+
+    long_integration_timeout = Instrument.control(
+        ":SENS<ch>:LINT:TOUT?", ":SENS<ch>:LINT:TOUT %g",
+        """A floating point property that controls the long integration timeout
+        in seconds, which takes values between 1 and 63. """,
+        validator=truncated_range,
+        values=[1, 63],
+    )
+
+    long_integration_fast_enabled = Instrument.control(
+        ":SENS<ch>:LINT:FAST?", ":SENS<ch>:LINT:FAST %d",
+        """A boolean property that controls whether long integration fast 
+        readings are enabled. """,
+        validator=strict_discrete_set,
+        values={True: 1, False: 0},
+        map_values=True
+    )
+
+    long_integration_search_enabled = Instrument.control(
+        ":SENS<ch>:LINT:SEAR?", ":SENS<ch>:LINT:SEAR %d",
+        """A boolean property that controls whether long integration search 
+        is enabled. """,
+        validator=strict_discrete_set,
+        values={True: 1, False: 0},
+        map_values=True
+    )
+
+    long_integration_detect_enabled = Instrument.control(
+        ":SENS<ch>:LINT:DET?", ":SENS<ch>:LINT:DET %d",
+        """A boolean property that controls whether long integration detection 
+        mode is enabled. """,
+        validator=strict_discrete_set,
+        values={True: 1, False: 0},
+        map_values=True
+    )
+
+    source_voltage = Instrument.control(
+        ":SOUR<ch>:VOLT?", ":SOUR<ch>:VOLT %g",
+        """A floating point property that controls the source voltage in volts, 
+        which takes values between 0 and 15. """,
+        validator=truncated_range,
+        values=[0, 15],
+    )
+
+    source_voltage_protection = Instrument.control(
+        ":SOUR<ch>:VOLT:PROT?", ":SOUR<ch>:VOLT:PROT %g",
+        """A floating point property that controls the source voltage protection 
+        offset in volts, which takes values between 0 and 8. """,
+        validator=truncated_range,
+        values=[0, 8],
+    )
+
+    source_voltage_protection_enabled = Instrument.measurement(
+        ":SOUR<ch>:VOLT:PROT:STAT?",
+        """A boolean property that returns the source voltage protection state. """
+    )
```

Comment:
Is this returning a boolean ? I think you could pass the parameter `cast=bool` to make a casting.
How do you enable/disable the voltage protection ?
Similar comment for `source_current_limit_enabled`.

Reply:
Good catch... it is not:

```python
INSTR.ch(1).source_voltage_protection_enabled
0.0
```

Will add a test and fix.

Reply:
@mfikes looking a the instrument manual, I understand this property is probably set when voltage protection has triggered. I infer this from user manual notes on the other similar property `source_current_limit_enabled  ` which allow to understand if the current limit has been reached.
Initially I thought, looking at doc string, these properties were indicating whether the user has enabled or not the specific protection.
Am I correct in this ? Do you have the same understanding ?

Reply:
@msmttchr Indeed, this is essentially a property that evaluates to `True` when voltage protection has triggered. (And in that case the voltage should have been shut off by the instrument.) And indeed, this—and the analogous current limit property—are not settable by the user. 

Unfortunately, for `source_current_limit_enabled`, the explanation is a little more complex because when it evaluates to `True` this could mean that the source is actually being current-limited or the source has been disabled (the current protection has _tripped_, in Keithley's terminology). Given this, the name of the property feels less than ideal. In the SCPI portion of the manual, the full command is `SOURCE:CURRENT:LIMIT:STATE?` where `1` means that the output is in limit mode or has tripped, depending on what mode the user put the device in.

So, I agree, these properties names and/or doc strings could be improved. I looked around for other PSU instruments in PyMeasure and didn't find anything like this being exposed, so perhaps this is a chance to set a good precedent.

Does the use of `_enabled` in a property name imply something that can be set (with some other naming being used for properties that are read-only)? (I had only chosen `_enabled` to go along with the fact that the property is Boolean-valued, no more rationale than that.)

In any case, it seems easy to expand upon the doc string to further clarify the meanings of each of these properties, perhaps something along these lines: 

Voltage:

"A boolean property that returns the source voltage protection state. If this property is True, the source has been shut off in accordance with the source voltage protection settings. If this property is False, the source has not been shut off due to voltage protection."

Current:

"A boolean property that returns the source current limit state. If this property is True, the source is in either in current limit mode, or has tripped (shut off), based on the source_current_limit_type setting. If this property is False, the source is not being limited and has not been tripped."

Reply:
@mfikes I am not aware of any convention for _enabled suffix, so I would not suggest to make any name change.
Regarding the clarification on the doc string, I agree with your proposal.

Reply:
Doc strings updated.

Reply:
> Does the use of _enabled in a property name imply something that can be set (with some other naming being used for properties that are read-only)? (I had only chosen _enabled to go along with the fact that the property is Boolean-valued, no more rationale than that.)

There is no rationale for naming indicating settability -- your chosen rationale was correct. In the end, we're happy if people choose clear names, and adjective-like names for boolean properties typically fit well and are clear (if you imagine them being a question "is power enabled?" or command "make power enabled")